rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/Serialization.kt

```diff
@@ -54,14 +56,44 @@ fun List<N_Lecture>.toJson() = LectureList(this.mapNotNull {
     }
 })
 
+/**
+ * A data class to represent the type of change that was sent/received by the websocket
+ */
 enum class WhatChanged {
     ROOM,
     ROOM_LIST,
     LECTURE,
     LECTURE_LIST
 }
-
+/**
+ * A data class to represent the change notification sent/received by the update websocket
+ */
 data class ChangeNotification(
-    val whatChanged: WhatChanged,
-    val change: Any
+        val whatChanged: WhatChanged,
+        val change:  Any
 )
+/**
+ * A Gson deserializer which can distinguish between the different WhatChanged types.
+ * Will return the deserialized ChangeNotification
+ */
+class ChangeNotificationDeserializer() : JsonDeserializer<ChangeNotification> {
+
+    private var gson: Gson = Gson()
+    private var changeTypeRegistry: MutableMap<WhatChanged, Class<out Any?>> = HashMap()
+    private var changeTypeElementName: String = "whatChanged"
+
+    fun registerChangeType(changeTypeName: WhatChanged, changeType: Class<out Any?>) {
+        changeTypeRegistry[changeTypeName] = changeType
+    }
+
+    override fun deserialize(json: JsonElement?, typeOfT: Type?, context: JsonDeserializationContext?): ChangeNotification {
+        val changeNotificationObject: JsonObject = json!!.asJsonObject
+
+        return when (val whatChanged: WhatChanged = WhatChanged.valueOf(changeNotificationObject[changeTypeElementName].asString)) {
```

Comment:
```suggestion
        val changeNotificationObject = json!!.asJsonObject

        return when (val whatChanged = WhatChanged.valueOf(changeNotificationObject[changeTypeElementName].asString)) {
```
Types can be inferred.